src/map/utils/mobutils.cpp

```diff
@@ -55,30 +55,155 @@ namespace mobutils
+            case REGION_TYPE::BASTOK:
+            case REGION_TYPE::WINDURST:
+            case REGION_TYPE::JEUNO:
+            case REGION_TYPE::DYNAMIS:
+            case REGION_TYPE::TAVNAZIAN_MARQ:
+            case REGION_TYPE::PROMYVION:
+            case REGION_TYPE::LUMORIA:
+            case REGION_TYPE::LIMBUS:
+                offset       = 2;
+                rangedOffset = 5;
+                break;
+            // TOAU Regions
+            case REGION_TYPE::WEST_AHT_URHGAN:
+            case REGION_TYPE::MAMOOL_JA_SAVAGE:
+            case REGION_TYPE::HALVUNG:
+            case REGION_TYPE::ARRAPAGO:
+            case REGION_TYPE::ALZADAAL:
+                offset       = 10;
+                rangedOffset = 12;
+                break;
+
+            // WOTG Regions
+            case REGION_TYPE::RONFAURE_FRONT:
+            case REGION_TYPE::NORVALLEN_FRONT:
+            case REGION_TYPE::GUSTABERG_FRONT:
+            case REGION_TYPE::DERFLAND_FRONT:
+            case REGION_TYPE::SARUTA_FRONT:
+            case REGION_TYPE::ARAGONEAU_FRONT:
+            case REGION_TYPE::FAUREGANDI_FRONT:
+            case REGION_TYPE::VALDEAUNIA_FRONT:
+                offset       = 11;
+                rangedOffset = 13;
+                break;
+
+            // Other
+            case REGION_TYPE::ABYSSEA:
+            case REGION_TYPE::THE_THRESHOLD:
+            case REGION_TYPE::ABDHALJS: // TODO: Need data for ABDHALJS zones.
+                offset       = 11;
+                rangedOffset = 13;
+                break;
+
+            // SOA Regions
+            case REGION_TYPE::ADOULIN_ISLANDS:
+            case REGION_TYPE::EAST_ULBUKA:
+                offset       = 11; // TODO: Need more data for Lvl 100+ mobs.
+                rangedOffset = 13;
+                break;
+
+            // Default Fallback
+            default:
+                offset       = 2;
+                rangedOffset = 5;
+                break;
+        }
+
+        offset += PMob->getMobMod(MOBMOD_DAMAGE_OFFSET);
 
         if (slot == SLOT_RANGED)
         {
-            bonus = 5;
+            offset = rangedOffset;
+            offset += PMob->getMobMod(MOBMOD_RANGED_DAMAGE_OFFSET);
         }
 
-        if (lvl == 1)
+        // Normal mobs in beginner zones have the offset lowered by 1.
+        // Excluded NMs for now for things like Voidwatch Mobs.
+        if (mobZoneId != 0 && PMob->m_Type != MOBTYPE_NOTORIOUS && (mobZoneId == ZONE_WEST_RONFAURE || mobZoneId == ZONE_EAST_RONFAURE || mobZoneId == ZONE_NORTH_GUSTABERG || mobZoneId == ZONE_SOUTH_GUSTABERG || mobZoneId == ZONE_WEST_SARUTABARUTA || mobZoneId == ZONE_EAST_SARUTABARUTA))
         {
-            bonus = 0;
+            offset -= 1;
         }
 
-        damage = lvl + bonus;
-
-        damage = (uint16)(damage * PMob->m_dmgMult / 100.0f);
+        // Clamp to 0 for edge cases that might cause the offset go negative.
+        if (offset < 0)
+        {
+            offset = 0;
+        }
```

Comment:
Arent this 2 checks out of order?
<img width="694" height="325" alt="imagen" src="https://github.com/user-attachments/assets/bf79d115-bfa3-463a-a48b-80f0b881c838" />

1st, under X conditions, you are setting offset to be < 0
Then you are clamping it